### PR TITLE
Tightening up the eventing publisher

### PR DESCRIPTION
**Pull request by obelisk**
Tighten up the event tapping event publisher code. This should hopefully make the code more stable and better equip it to handle edge cases.

### CODE REVIEW

File: osquery/events/darwin/event_taps.cpp [hunk 1]
```diff
@@ -26,9 +27,9 @@ CGEventRef EventTappingEventPublisher::eventCallback(CGEventTapProxy proxy,
                                                      CGEventType type,
                                                      CGEventRef event,
                                                      void* refcon) {
-  auto ec = createEventContext();
-  EventFactory::fire<EventTappingEventPublisher>(ec);
-  return event;
+  EventFactory::fire<EventTappingEventPublisher>(createEventContext());
+  /// If you change from listenOnly, return event or you will drop all events
```
Comment (theopolis) [inline osquery/events/darwin/event_taps.cpp]: You don't need `///` just `//`

File: osquery/events/darwin/event_taps.cpp [hunk 2]
```diff
@@ -43,39 +44,45 @@ void EventTappingEventPublisher::tearDown() {
...
 
 void EventTappingEventPublisher::stop() {
-  WriteLock lock(run_loop_mutex_);
+  WriteLock lock(runLoopMutex_);
 
-  if (run_loop_source_ != nullptr) {
-    CFRunLoopRemoveSource(run_loop_, run_loop_source_, kCFRunLoopCommonModes);
-    CFRelease(run_loop_source_);
-    run_loop_source_ = nullptr;
+  if (runLoopSource_ != nullptr) {
```
Comment (theopolis) [inline osquery/events/darwin/event_taps.cpp]: You should use snake case for local member variables, but we don't enforce it.

File: osquery/events/darwin/event_taps.cpp [hunk 3]
```diff
@@ -43,39 +44,45 @@ void EventTappingEventPublisher::tearDown() {
...
-  if (eventTap == nullptr) {
+  WriteLock lock(runLoopMutex_);
+
+  runLoop_ = CFRunLoopGetCurrent();
+  eventTap_ = CGEventTapCreate(kCGSessionEventTap,
+                              kCGHeadInsertEventTap,
+                              kCGEventTapOptionListenOnly,
+                              (1 << kCGEventKeyDown),
+                              eventCallback,
+                              NULL);
```
Comment (theopolis) [inline osquery/events/darwin/event_taps.cpp]: Should the `NULL` be a `nullptr`?

File: osquery/events/darwin/event_taps.cpp [hunk 4]
```diff
@@ -43,39 +44,45 @@ void EventTappingEventPublisher::tearDown() {
...
+    CFRunLoopRemoveSource(runLoop_, runLoopSource_, kCFRunLoopCommonModes);
+    CFRelease(runLoopSource_);
+    runLoopSource_ = nullptr;
   }
-  if (run_loop_ != nullptr) {
-    CFRunLoopStop(run_loop_);
-    run_loop_ = nullptr;
+
+  if (eventTap_ != nullptr) {
+    CGEventTapEnable(eventTap_, false);
```
Comment (theopolis) [inline osquery/events/darwin/event_taps.cpp]: Should you do this "disable" before removing the source? Not sure, just curious?

**Comment by osqueryer** [comment]
:no_good: The commit 6bf2afabdcedaf9c1afd798b19d5124bb251aebf (Job results: [2788](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2788/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit 2dd3222330fe69e9ed282f39e225124678940a55 (Job results: [2789](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2789/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit 5c8e5cff6691b8a3df7b497e82faa5bb8282f062 (Job results: [2790](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2790/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 2ae754f51a8d7577e163377f7028ed4dc1098931 (Job results: [3041](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3041/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 2ae754f51a8d7577e163377f7028ed4dc1098931 (Job results: [3042](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3042/)) failed one or more tests (**Windows**).

**Pull request merged**